staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -139,6 +139,35 @@ func NewAvailableConditionController(
 	return c
 }
 
+func setConditionAndUpdateStatus(client apiregistrationclient.APIServicesGetter, apiService *apiregistration.APIService, newCondition apiregistration.APIServiceCondition) error {
+	orig := apiService.DeepCopy()
+	apiregistration.SetAPIServiceCondition(apiService, newCondition)
+	if reflect.DeepEqual(orig, apiService) {
+		return nil
+	}
+
+	_, err := client.APIServices().UpdateStatus(apiService)
+	if err != nil {
+		return err
+	}
+
+	// update metrics
+	isAvailable := apiregistration.IsAPIServiceConditionTrue(apiService, apiregistration.Available)
+	if isAvailable {
+		unavailableGauge.WithLabelValues(apiService.Name).Set(0.0)
+	} else {
+		unavailableGauge.WithLabelValues(apiService.Name).Set(1.0)
+	}
+	if newCondition.Type == apiregistration.Available {
```

Comment:
I believe this check may be unnecessary (if I am reading this correctly). Since we always pass in a single condition, the diff between the original and the currentAPIService should always be between a single condition, which means we can just check wasAvailable directly and compare that with isAvailable to increment the counter.

Reply:
fixed